tests/func/test_get.py

```diff
@@ -6,178 +6,185 @@
+
     with pytest.raises(UrlNotDvcRepoError):
-        Repo.get(git_erepo.root_dir, "some_file.zip")
+        Repo.get(fspath(erepo_dir), "some_file.zip")
 
 
-def test_get_a_dvc_file(erepo):
+def test_get_a_dvc_file(tmp_dir, erepo_dir):
     with pytest.raises(GetDVCFileError):
-        Repo.get(erepo.root_dir, "some_file.dvc")
+        Repo.get(fspath(erepo_dir), "some_file.dvc")
 
 
 # https://github.com/iterative/dvc/pull/2837#discussion_r352123053
-def test_get_full_dvc_path(erepo):
-    external_data_dir = erepo.mkdtemp()
-    external_data = os.path.join(external_data_dir, "ext_data")
-    with open(external_data, "w+") as fobj:
-        fobj.write("ext_data")
-
-    cur_dir = os.getcwd()
-    os.chdir(erepo.root_dir)
-    erepo.dvc.add(external_data)
-    erepo.dvc.scm.add(["ext_data.dvc"])
-    erepo.dvc.scm.commit("add external data")
-    os.chdir(cur_dir)
-
-    Repo.get(erepo.root_dir, external_data, "ext_data_imported")
-    assert os.path.isfile("ext_data_imported")
-    assert filecmp.cmp(external_data, "ext_data_imported", shallow=False)
-
-
-def test_non_cached_output(tmp_path, erepo):
-    os.chdir(erepo.root_dir)
-    erepo.dvc.run(
-        outs_no_cache=["non_cached_file"], cmd="echo hello > non_cached_file"
+def test_get_full_dvc_path(tmp_dir, erepo_dir, tmp_path_factory, monkeypatch):
+    path = tmp_path_factory.mktemp("ext")
+    external_data = path / "ext_data"
+    external_data.write_text("ext_data")
+
+    with monkeypatch.context() as m:
+        m.chdir(fspath(erepo_dir))
+        erepo_dir.dvc.add(fspath(external_data))
+        erepo_dir.scm_add(["ext_data.dvc"], commit="add external data")
+
+    Repo.get(fspath(erepo_dir), fspath(external_data), "ext_data_imported")
+    assert (tmp_dir / "ext_data_imported").is_file()
+    assert (tmp_dir / "ext_data_imported").read_text() == "ext_data"
+
+
+def test_non_cached_output(tmp_dir, erepo_dir):
+    src = "non_cached_file"
+    dst = src + "_imported"
+
+    erepo_dir.dvc.run(
+        outs_no_cache=[src],
+        cmd="echo hello > non_cached_file",
+        cwd=fspath(erepo_dir),
     )
-    erepo.dvc.scm.add(["non_cached_file", "non_cached_file.dvc"])
-    erepo.dvc.scm.commit("add non-cached output")
-    os.chdir(fspath_py35(tmp_path))
-    Repo.get(erepo.root_dir, "non_cached_file")
+    erepo_dir.scm.add([src, src + ".dvc"])
+    erepo_dir.scm.commit("add non-cached output")
 
-    src = os.path.join(erepo.root_dir, "non_cached_file")
-    assert os.path.isfile("non_cached_file")
-    assert filecmp.cmp(src, "non_cached_file", shallow=False)
+    Repo.get(fspath(erepo_dir), src, dst)
+
+    assert (tmp_dir / dst).is_file()
+    # NOTE: using strip() to account for `echo` differences on win and *nix
+    assert (tmp_dir / dst).read_text().strip() == "hello"
 
 
 # https://github.com/iterative/dvc/pull/2837#discussion_r352123053
-def test_absolute_file_outside_repo(erepo):
+def test_absolute_file_outside_repo(tmp_dir, erepo_dir):
```

Comment:
So we wasn't using `tmp_dir` before. We simply made a get into a current dir? And it wasn't polluting only because nothing was written prior to error?

Reply:
@Suor Yes, and in case something went wrong we would be polluting cwd, which is bad. So I've added that one.